modules/images/image-loading-optimization/optimization.php

```diff
@@ -107,7 +108,12 @@ function ilo_optimize_template_output_buffer( string $buffer ): string {
 	);
 
 	// Whether we need to add the data-ilo-xpath attribute to elements and whether the detection script should be injected.
-	$needs_detection = ilo_needs_url_metric_for_breakpoint( $needed_minimum_viewport_widths );
+	$needs_detection = in_array(
+		true,
+		// Each array item is array{int, bool}, with the second item being whether the viewport width is needed.
+		array_column( $needed_minimum_viewport_widths, 1 ),
+		true
+	);
```

Comment:
Some cool things with PHPStan's static analysis here. If 0 is supplied as the second arg to `array_column()` it detects an error:

> phpstan: Call to function in_array() with arguments true, list<int> and true will always evaluate to false.

This is because from the array shape returned by `ilo_get_needed_minimum_viewport_widths()` it knows that the first column will all be integers (the minimum viewport widths) and thus the `in_array()` check will always be `false`. 

Conversely, if I supply a higher index like 2 in `array_column()`, I get:

> phpstan: Call to function in_array() with arguments true, array{} and true will always evaluate to false.

Here PHPStan knows that since there are only two items in each tuple, the return of `array_column()` will be an empty array, and thus `in_array()` will also always return `false`.